pubpub/pubpub
# feat: add instagram, mastodon, linkedin, bluesky and github socials for community and users, and make social navbar configurable

## Issue(s) Resolved

Closes #2514 
Closes #2513 

## Test Plan

1. Go to `/dash/settings/details`
2. Add an instagram and mastodon profile
3. Check that they show up in the footer and navbar
4. Check that they resolve to the proper page
5. For mastodon, check that it has the `rel="me"` attribute. This is used to verify that your domain belongs to you on mastodon

### Updated
1. go to `jtrialerror` community
2. click every link in the social navbar
3. Check that they go to a valid profile
4. go to `/user/profile/thomas-jorna`
5. click every link in that navbar
6. check that they go to a valid profile
7. Go to /dash/settings
8. Change the "These links will appear in your Community's " setting to "header only."
9. Save
10. Check that on `/` the socials no longer appear in the footer
11. Change it to "footer only"
12. Check that on `/` the socials no longer appear in the header
13. Change it back to `both`
14. Make screen width < 700px
15. Observe that social header dissapears even if setting is "header" or "both"

## Screenshots (if applicable)


New config
<img width="497" alt="image" src="https://github.com/pubpub/pubpub/assets/21983833/269039c8-1abc-4962-b7b8-eb93215eaf51">


## Optional



### Notes/Context/Gotchas

> [!WARNING]
> Migration needs to be run on prod for this to work

> [!NOTE]
> I hid the social navbar menu on mobile view, as it became almost impossible to see the normal navbar if all social items were set.
> I think it somewhat unnecessary to have the socials in the upper navbar, takes up too much space and is already available in the footer. 


Header only setting is maybe a bit redundant, as probably no one is going to set it to that as it will be hidden in mobile view.


### Possible follow ups

1. [x] Also add insta/mastodon to profile pages
3. [x] Replace facebook icon with newer one (the round one, the square one is outdated)
5. Replace twitter logo/name with X (bleh)

### Supporting Docs
